### DISCUSSION TITLE
should have an option to pop the history on set

### DISCUSSION THREAD

**Discussion by jebie** [47ng/nuqs#528] [Ideas]
I'm using this modal and acts as a new page, but the downside on mobile view is that when the back button on mobile is clicked, it goes back to the previous page instead of exiting the modal.

So, what happens is, I set the history to "push" and it works well, but the downside is that when the modal is closed, the pushed URL remains. Consequently, when clicking the back button, it takes two clicks to go back to the previous page.

It would be nice to have an option to pop the history. something like this.

```js
setCloseModal(false, { history: "pop" });
```

**Comment by franky47** [comment]
There are a few questions that need answering here:

1. What should happen if the user loads the page from `/path/to/page?modal=true`, then either:
  a. Closes the modal via the UI, then uses the back button
  b. Uses the back button directly

You could wire the UI control in charge of closing the modal to the `history.back` method, which would behave the same as hitting the back button, and act like a history pop (which doesn't exist in the [History API](https://developer.mozilla.org/en-US/docs/Web/API/History_API/Working_with_the_History_API)).

The issue being that in case 1.a, closing the modal will go back to the URL the user was on before landing in your app. You could detect this using a local React state (or the absence thereof) when you open the modal from within your app.
